cpg_infra/abstraction/hailbatch.py

```diff
@@ -31,7 +31,17 @@
 )
 
 
-def get_hail_batch_token(token_category) -> str:
+def get_hail_batch_auth_headers(token_category: str, batch_uri: str) -> dict[str, str]:
+    token = get_hail_batch_auth_token(token_category)
+    headers = {'Authorization': f'Bearer {token}'}
+    # If this is a dev service then need an extra header
+    if batch_uri.startswith('https://internal.hail'):
+        headers['X-Hail-Internal-Authorization'] = f'Bearer {token}'
```

Comment:
Does this work? My understanding was these are actually different auth tokens [team-docs/hail.md](https://github.com/populationgenomics/team-docs/blob/main/hail.md#:~:text=This%20now%20requires%20two%20tokens%3A%20one%20for%20the%20default%20namespace%2C%20and%20the%20other%20for%20your%20development%20namespace.%20They%20need%20to%20be%20passed%20as%20separate%20headers%3A)?

Reply:
Yeah it does work, I think because I'm using GCP credentials the same credentials work for dev and prod. I'm not sure if there's another type of token that would require different ones for dev and prod?

Reply:
Cool! There are session tokens you can provide (the one you created and put in tmp) and supply those. That's actually how the billing aggregator works in prod (there's an environment variable) for each environment. 

Not to say we shouldn't move that over to GCP / Azure credentials to do that :) [but maybe for later]